api/nakadi-event-bus-api.yaml

```diff
@@ -0,0 +1,1232 @@
+            $ref: '#/definitions/Problem'
+        '403':
+          description: Not allowed
+          schema:
+            $ref: '#/definitions/Problem'
+        '404':
+          description: |
+            Not found.
+          schema:
+            $ref: '#/definitions/Problem'
+        '422':
+          description: Unprocessable Entity
+          schema:
+            $ref: '#/definitions/Problem'
+        '500':
+          description: Server error
+          schema:
+            $ref: '#/definitions/Problem'
+
+        '503':
+          description: Not available
+          schema:
+            $ref: '#/definitions/Problem'
+
+  '/registry/validation-rule':
+    get:
+      tags:
+        - schema-registry-api
+      description: |
+        Lists all of the validation rules supported by this installation of Nakadi.
+
+        If the EventType creation is to have special validations (besides the default), one can consult over
+        this method the available possibilities.
+      responses:
+        '200':
+          description: Returns a list of all `EventValidationStrategy` known to Nakadi
+          schema:
+            type: array
+            items:
+              $ref: '#/definitions/EventValidationStrategy'
+
+
+# ################################### #
+#                                     #
+#             Definitions             #
+#                                     #
+# ################################### #
+
+definitions:
+  Event:
+    type: object
+    description: |
+      **Note** The Event definition will be externalized in future versions of this document.
+
+      A basic payload of an Event. The actual schema is dependent on the information configured for the
+      EventType, as is its enforcement (see POST /event-types). Setting of metadata properties are
+      dependent on the configured enrichment as well.
+
+      For explanation on default configurations of validation and enrichment, see documentatin of
+      `EventType.type`.
+
+      For concrete examples of what will be enforced by Nakadi see the objects BusinessEvent and
+      DataChangeEvent below.
+    additionalProperties: true
+
+
+  EventMetadata:
+    type: object
+    description: |
+      Metadata for this Event.
+
+      Contains commons fields for both Business and DataChange Events. Most are enriched by Nakadi upon
+      reception, but they in general MIGHT be set by the client.
+    properties:
+      metadata:
+        type: object
+        properties:
+          eid:
+            description: |
+              Identifier of this Event.
+
+              Clients are allowed to generate this and this SHOULD be guaranteed to be unique from the
+              perspective of the producer. Consumers MIGHT use this value to assert uniqueness of reception
+              of the Event.
+
+              If not present upon reception, Nakadi will generate and fill this with a unique value.
+              (TBD: enforce UUID formatif set by the client?)
```

Comment:
IMO it must be provided by producer -- and producer has to take care that it is unique, e.g. using UUIDs.
There always is the (low) risk that event duplicates may be created.
And generally clients should have the chance to detect that duplicates and handle it.
If eid is generated by Nakadi than producer event POST retries (e.g. after connection failure) will generate duplicates and client has no chance to do the de-duplication.
Should be described here too.